app/domain/operations/hbx_admin/dry_run/notice_query.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Operations
+  module HbxAdmin
+    module DryRun
+      # This Operation is responsible for building a query to fetch notice counts based on given parameters.
+      class NoticeQuery
+        include Dry::Monads[:do, :result]
+        include L10nHelper
+
+        # Calls the NoticeQuery operation.
+        #
+        # @param params [Hash] The parameters for the query.
+        # @option params [Array<String>] :person_hbx_ids The person HBX IDs.
+        # @option params [Date] :start_date The start date for the query.
+        # @option params [Date] :end_date The end date for the query.
+        # @option params [Array<String>] :title_codes The title codes for the query.
+        # @return [Dry::Monads::Result] The result of the operation.
+        def call(params)
+          validated_params = yield validate(params)
+          build_pipeline(validated_params)
+        end
+
+        private
+
+        # Validates the input parameters.
+        #
+        # @param params [Hash] The parameters to validate.
+        # @return [Dry::Monads::Result] The result of the validation.
+        def validate(params)
+          return Failure("Titles cannot be blank") if params[:title_codes].blank?
+          return Failure("Person HBX IDs cannot be blank") if params[:person_hbx_ids].blank?
+          return Failure("Start Date cannot be blank") if params[:start_date].blank?
+          return Failure("End Date cannot be blank") if params[:end_date].blank?
```

Comment:
Missing translations as these will be presented in the UI.

Reply:
these are not displayed in UI.